src/s3_fetchers.rs

```diff
@@ -160,3 +220,89 @@ async fn fetch_shard_or_retry(
         near_indexer_primitives::IndexerShard,
     >(body_bytes.as_ref())?)
 }
+
+#[cfg(test)]
```

Comment:
Can we move tests into a separate file(s) as a basement for future tests, please?

Reply:
My understanding was that convention was to place unit tests within the same file so you are able to test both public and private methods. Maybe one day we'll want to test `fetch_shard_or_retry` in isolation, which is currently private. It might be best to keep it this way, thoughts?